plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/HiveS3Config.java

```diff
@@ -60,6 +60,7 @@
     private Duration s3MaxBackoffTime = new Duration(10, TimeUnit.MINUTES);
     private Duration s3MaxRetryTime = new Duration(10, TimeUnit.MINUTES);
     private Duration s3ConnectTimeout = new Duration(5, TimeUnit.SECONDS);
+    private Duration s3connectTtl = new Duration(60, TimeUnit.MINUTES);
```

Comment:
60 minuts seems pretty high. Could it cause queries to hang?

Reply:
> 60 minuts seems pretty high. Could it cause queries to hang?

@sopel39 Good question, but we have used S3 in several trino clusters for several years, and didn't meet the problem described in the issue until these days. The default value from Amazon is -1, which means the connection will never expire, that is much longer than 60min.  

When the network to S3 is stable, and the number of connections is not too large, the data fetching performance is stable, so the HTTP error mentioned above only happens in a enough high load scenario.  

We set the default value to 60 min, because we saw that the Nexus(a Maven repository manager) community said that they set the ttl value to 3600s: https://issues.sonatype.org/browse/NEXUS-28266  

And we thought that most queries(even some ETL big tasks) finishes within 1 hour, so maybe 1 hour is a suitable default value in the production environment(which is also used in ours).  

I can change the default value to another one if our community has a better advice value~

Reply:
Personally, I'd suggest the default option to cause no behavior change over the current implementation. So if the config option is not set the [DEFAULT_CONNECTION_TTL](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/ClientConfiguration.html#DEFAULT_CONNECTION_TTL) is used

Reply:
> Personally, I'd suggest the default option to cause no behavior change over the current implementation. So if the config option is not set the [DEFAULT_CONNECTION_TTL](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/ClientConfiguration.html#DEFAULT_CONNECTION_TTL) is used

I understand your concerns~But the default ttl is -1, it means that the connection will never expire and always be reused by S3 client. We asked the AWS support team, they suggested to reduce this ttl value.

When the cluster load is high enough, or there are many connections to S3 server, or the network to S3 is not stable, or the S3 server from AWS actively closed the connection, the never expired connection objects in the sdk client will cause some border issues.

Reply:
> Personally, I'd suggest the default option to cause no behavior change over the current implementation. So if the config option is not set the [DEFAULT_CONNECTION_TTL](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/ClientConfiguration.html#DEFAULT_CONNECTION_TTL) is used

@alexjo2144 I think you are right, we should give the choice of whether to adjust the ttl value to users, maybe there is no perfect default value for all environments, and maintaining Amazon's official default value is the safest choice.

Because if other users encounter similar problems to ours, they will search and see our adjustment suggestions in the issue above, lol.  

I have changed the default ttl value in `HiveS3Config.java` to [DEFAULT_CONNECTION_TTL](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/ClientConfiguration.html#DEFAULT_CONNECTION_TTL), so the commit will not change the current implementation, and should be safe to be merged now, I saw all the tests and checks passed last time. 

Please review again, thank you.

Reply:
> Personally, I'd suggest the default option to cause no behavior change over the current implementation. So if the config option is not set the [DEFAULT_CONNECTION_TTL](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/ClientConfiguration.html#DEFAULT_CONNECTION_TTL) is used

@alexjo2144 You are right, we should give the choice of whether to adjust the ttl value to users, maybe there is no perfect default value for all environments, and maintaining Amazon's official default value is the safest choice.  

Because if other users encounter similar problems to ours, they will search and see our adjustment suggestions in the issue above, lol.  

I have changed the default ttl value to [DEFAULT_CONNECTION_TTL](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/ClientConfiguration.html#DEFAULT_CONNECTION_TTL), so the commit will not change the current implementation.  

Please review again, thank you. I have to use Optional value in `HiveS3Config.java` because `Duration` can't accept negative values such as -1.